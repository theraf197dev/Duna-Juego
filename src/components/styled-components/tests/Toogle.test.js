import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Toogle from '../components/Toggle';

describe('Toogle tests', () => {
  let props = {
    handleChange: jest.fn(),
    id: 'testId',
    value: false,
  };

  it('render component', () => {
      const { container } = render(<Toogle {...props} />);

      expect(container).toMatchSnapshot();
  });

  it('toogleButton is clicked and his states changes', async () => {
      const testProps = {
        ...props,
        handleChange: jest.fn(),
      };

      render(<Toogle {...testProps} />);

      const toogleButton = await screen.findByTestId('fn-toogle-button');
      userEvent.click(toogleButton);

      expect(testProps.handleChange).toHaveBeenCalled();
  });
});
