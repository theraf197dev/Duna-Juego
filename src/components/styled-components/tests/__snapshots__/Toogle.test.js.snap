// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toogle tests render component 1`] = `
<div>
  <div
    class="sc-bcXHqe hEUNKZ"
  >
    <input
      class="sc-gswNZR gNWEtq"
      data-testid="fn-toogle-button"
      id="testId"
      type="checkbox"
    />
    <label
      class="sc-dkrFOg hBVbHy"
      for="testId"
    >
      <div
        class="sc-hLBbgP kxibuV"
      >
        <div
          class="sc-eDvSVe rLpUP"
        >
          ★
        </div>
        <div
          class="sc-jSUZER hDTQsy"
        >
          ★
        </div>
      </div>
      <div
        class="sc-gKPRtg cFVFRL"
      />
    </label>
  </div>
</div>
`;
